File: ruby/src/Creational/Prototype/PrototypeFactory.rb

#//Prototype Overview
#//Make new objects by cloning the objects which you set as prototypes
#//Still reading? Save your time, watch the video lessons!
#//Video tutorial on design patterns

# PrototypeFactory - a Factory for Prototypes

class PrototypeFactory

	def initialize(prototypeSpoon,prototypeFork)		
		@prototypeSpoon = prototypeSpoon
		@prototypeFork = prototypeFork
	end
	
	def makeSpoon()
 		@prototypeSpoon.clone
	end

	def makeFork()
 		@prototypeFork.clone
	end

end

File: scala/src/main/scala/Creational/Prototype/PrototypeFactory.scala
package Creational.Prototype;

//// Original copy of this content taken from http://www.fluffycat.com/Java-Design-Patterns/ in 2010
//// Original Author: Larry Truett
//// Privacy Policy at http://www.fluffycat.com/Privacy-Policy/
//Java Design Patterns Prototype
//Prototype Overview
//Make new objects by cloning the objects which you set as prototypes
//Still reading? Save your time, watch the video lessons!
//Video tutorial on design patterns
//PrototypeFactory - a Factory for Prototypes

public class PrototypeFactory {  
    AbstractSpoon prototypeSpoon;
    AbstractFork prototypeFork;
    
    public PrototypeFactory(AbstractSpoon spoon, AbstractFork fork) {
       prototypeSpoon = spoon;
       prototypeFork = fork;
   }
    
   public AbstractSpoon makeSpoon() {
       return (AbstractSpoon)prototypeSpoon.clone();
   }
   public AbstractFork makeFork() {
       return (AbstractFork)prototypeFork.clone();
   }
}